control/grpc.py

```diff
@@ -329,30 +329,19 @@ def create_listener(self, request, context=None):
         self.logger.info(f"Received request to create {request.gateway_name}"
                          f" {request.trtype} listener for {request.nqn} at"
                          f" {request.traddr}:{request.trsvcid}.")
+        if request.gateway_name != self.gateway_name:
```

Comment:
This is not correct. If a request is not for us but for some other valid gateway, we should process it, i.e. store in omap and notify other gateways in the group to update its state.

So you should sheck if the name in the request is in the list of valid (existing) gateways, but currently we don't have such a list and it is not clear if we want this restriction actually. As I said previously this may be out of the scope of this PR. Your previous version looked good enough for me.

Reply:
Sorry, I misunderstood. Thank you for explaining and gaining a deeper understanding. Restore to previous version.